pkg/model/job.go

```diff
@@ -137,6 +139,11 @@ type LabelSelectorRequirement struct {
 	Values []string `json:"Values,omitempty"`
 }
 
+type PublisherSpec struct {
+	Type   Publisher              `json:"Type,omitempty"`
+	Params map[string]interface{} `json:"Params,omitempty"`
```

Comment:
While I understand it will add more flexibility, I doubt in practice it will be needed and can add more complexity to handling our spec. I am also trying not to deviate too much from [IPVM task schema](https://github.com/ucan-wg/invocation#23-ipld-schema)